lib/member.symphony.php

```diff
@@ -372,6 +372,80 @@ public function filter_LockActivation(array &$context) {
 		}
 
 		/**
+		 * Validate Password
+		 * If there is an authentication field, we can check if a posted password is
+		 * valid for the currently logged-in Member. This filter is supposed to be
+		 * called as a pre-save filter, so it can terminate a section event (thus
+		 * prevent saving) if no valid password is provided.
+		 *
+		 * @param array $context
+		 */
+		public function filter_ValidatePassword(array &$context) {
+			$member = $this->Member;
+			if (!$member instanceof Entry) {
+				$context['messages'][] = array(
+					'member-validate-password',
+					false,
+					__('Member not found.'),
+					array(
+						'message-id' => MemberEventMessages::MEMBER_INVALID
+					)
+				);
+
+				return;
+			}
+
+			if (is_null($this->section->getFieldHandle('authentication'))) {
```

Comment:
We could do ! instead of is_null ?

Reply:
By design `getFieldHandle` will return `null` or a string, so why do you prefer `!`?

I'd prefer to be consistent with the rest of the code. If you search the extension for `$this->section->getFieldHandle`, you will see that `is_null` is always used to test the return value.

Reply:
So if we need to change how these return values are tested, I suggest to open another issue.

Reply:
Na it's ok.